src/interaction/inspector.jl

```diff
@@ -704,6 +704,16 @@ function show_imagelike(inspector, plot, name, edge_based)
         _pixelated_getindex(plot[1][], plot[2][], plot[3][], mpos, edge_based)
     end
 
+    # in case we hover over NaN values
+    if isnan(z)
+        a._visible[] = false
+        # return false as this plot is not processed (due to NaN) 
+        # and we have to process other plots from the figure
+        # see @show_data_recursion
+        return false
```

Comment:
This should still return true. In this case getting processed means that there is a `show_data` method that decided what to do with a hovered plot.